service_configuration_lib/spark_config.py

```diff
@@ -250,6 +250,35 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
     return spark_opts
 
 
+def _get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
+    if ('spark.dynamicAllocation.enabled' not in spark_opts or
+        str(spark_opts['spark.dynamicAllocation.enabled']) != 'true'):
+        return spark_opts
+
+    spark_opts['spark.dynamicAllocation.shuffleTracking.enabled'] = "true"
+    spark_opts['spark.dynamicAllocation.executorAllocationRatio'] = "0.8"
+
+    if 'spark.dynamicAllocation.minExecutors' not in spark_opts:
+
+        min_executors = int(spark_opts.get('spark.executor.instances', DEFAULT_EXECUTOR_INSTANCES)) // 4
```

Comment:
Are we sure every job is specifying `spark.executor.instances` at this stage?
I recall for some jobs it's calculated from spark.cores.max and spark.executor.cores. Is it before or after this?
I don't think using default value before checking makes sense.

Reply:
I would still add a warning or info for using DEFAULT_EXECUTOR_INSTANCES and recommend providing a manual one.

Reply:
Yes, the `spark.executor.instances` have been calculated at this stage. Should I add the warning for using DEFAULT_EXECUTOR_INSTANCES irrespective of DRA?

Reply:
Yes, please.